ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryAnonymizerTest.java

```diff
@@ -0,0 +1,89 @@
+import io.confluent.ksql.test.loader.JsonTestLoader;
+import io.confluent.ksql.test.tools.TestCase;
+import io.confluent.ksql.util.GrammarTokenExporter;
+import java.io.FileReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import org.approvaltests.Approvals;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import io.confluent.ksql.util.QueryAnonymizer;
+import org.junit.Before;
+import org.junit.Test;
+
+public class QueryAnonymizerTest {
+
+  private static final Path QUERIES_TO_ANONYMIZE_PATH =
+      Paths.get("src/test/java/io/confluent/ksql/test/QueriesToAnonymizeTest.txt");
+  private static final Path QUERY_VALIDATION_TEST_DIR = Paths.get("query-validation-tests");
+  private final QueryAnonymizer anon = new QueryAnonymizer();
+  private final Stream<TestCase> testCases = testFileLoader().load();
+  private List<String> sqlTokens;
+
+  @Before
+  public void setUp() {
+    sqlTokens = GrammarTokenExporter.getTokens();
+    sqlTokens.addAll(Arrays.asList("", "INT", "DOUBLE", "VARCHAR", "BOOLEAN", "BIGINT", "VALUE", "KEY"));
+  }
+
+  @Test
+  public void queriesAreAnonymizedCorrectly() throws Exception {
+    // Read queries
+    ArrayList<String> statements = new ArrayList<>();
+    try (Scanner s = new Scanner(new FileReader(QUERIES_TO_ANONYMIZE_PATH.toAbsolutePath().toString()))) {
+      while (s.hasNext()) {
+        statements.add(s.nextLine());
+      }
+    }
+
+    // anonymize statements
+    final List<String> anonStatementSelection = new ArrayList<>();
+    for (String statement : statements) {
+      anonStatementSelection.add(anon.anonymize(statement));
+    }
+
+    // Assert:
+    Approvals.verifyAll("Integration tests", anonStatementSelection);
+  }
+
+
+  @Test
+  public void anonQuerySplitByWordsHasOnlyTokensInSetIntersectionWithQuery() throws Exception {
+    // Given:
+    final Set<String> statements = new HashSet<>();
+    testCases
+        .filter(statement -> !statement.expectedException().isPresent())
+        .map(TestCase::statements)
+        .forEach(statements::addAll);
+
+    // When:
+    for (String statement : statements) {
+      final String anonStatement = anon.anonymize(statement);
+      final Set<String> statementWord = new HashSet<>(Arrays.asList(statement.split("[\\s\\(\\)<>,;]")));
+      final Set<String> anonStatementWord = new HashSet<>(Arrays.asList(anonStatement.split("[\\s\\(\\)<>,;]")));
+      final Set<String> intersection = new HashSet<>(statementWord);
+      intersection.retainAll(anonStatementWord);
+
+      // Assert:
+      Set<String> test = new HashSet<>(intersection);
+      intersection.removeAll(sqlTokens);
+      if (intersection.size() > 0) {
+        throw new Exception(String.format("Expected: %s\n Original: %s\n Wrong Tokens: %s",
+            statement, anonStatement, intersection.toString()));
+      }
+    }
```

Comment:
This test is going to be hard to debug, you should look at parametrizing it:
https://www.guru99.com/junit-parameterized-test.html
(First google result so I don't know how good)